JsonPatch.Tests/PatchExtensionTests.cs

```diff
@@ -0,0 +1,258 @@
+				"{\"op\":\"add\",\"path\":\"/Attributes/2/test1\",\"value\":\"test321\"},"+
+				"{\"op\":\"replace\",\"path\":\"/Attributes/3/test/2\",\"value\":3},"+
+				"{\"op\":\"add\",\"path\":\"/Attributes/4\",\"value\":{\"test\":[1,2,3]}}]";
+			
+			var patch = initial.CreatePatch(expected);
+			
+			Assert.AreEqual(patch.Operations.Count, 6);
+			Assert.AreEqual(patchExpected, JsonSerializer.Serialize(patch));
+			Assert.AreNotEqual(initial.Id, expected.Id);
+			Assert.AreNotEqual(initial.Attributes.RootElement.ToString(), expected.Attributes.RootElement.ToString());
+
+			var actual = patch.ApplyPatch(initial);
+
+			Assert.AreEqual(actual.Id, expected.Id);
+			Assert.AreEqual(actual.Attributes.RootElement.ToString(), expected.Attributes.RootElement.ToString());
+		}
+
+		[Test]
+		public void Add_Test()
+		{
+			var initial = new TestModel()
+			{
+				Id = Guid.NewGuid()
+			};
+			var expected = new TestModel()
+			{
+				Id = initial.Id,
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]")
+			};
+			var patchExpected = "[{\"op\":\"add\",\"path\":\"/Attributes\",\"value\":[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]}]";
+			
+			var patch = initial.CreatePatch(expected, new JsonSerializerOptions() {IgnoreNullValues = true});
+			
+			Assert.AreEqual(patch.Operations.Count, 1);
+			Assert.AreEqual(patch.Operations[0].Op, OperationType.Add);
+			Assert.AreEqual(patchExpected, JsonSerializer.Serialize(patch));
+
+			Assert.Null(initial.Attributes);
+			var actual = patch.ApplyPatch(initial);
+			Assert.AreEqual(actual.Attributes.RootElement.ToString(), expected.Attributes.RootElement.ToString());
+		}
+
+		[Test]
+		public void Remove_Test()
+		{
+			var initial = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]")
+			};
+			var expected = new TestModel()
+			{
+				Id = initial.Id
+			};
+			var patchExpected = "[{\"op\":\"remove\",\"path\":\"/Attributes\"}]";
+			
+			var patch = initial.CreatePatch(expected, new JsonSerializerOptions() {IgnoreNullValues = true});
+			
+			Assert.AreEqual(patch.Operations.Count, 1);
+			Assert.AreEqual(patch.Operations[0].Op, OperationType.Remove);
+			Assert.AreEqual(patchExpected, JsonSerializer.Serialize(patch));
+
+			Assert.NotNull(initial.Attributes);
+			var actual = patch.ApplyPatch(initial);
+			Assert.Null(actual.Attributes);
+		}
+
+		[Test]
+		public void Replace_Test()
+		{
+			var initial = new TestModel()
+			{
+				Id = Guid.NewGuid()
+			};
+			var expected = new TestModel()
+			{
+				Id = Guid.Parse("a299e216-dbbe-40e4-b4d4-556d7e7e9c35")
+			};
+			var patchExpected = "[{\"op\":\"replace\",\"path\":\"/Id\",\"value\":\"a299e216-dbbe-40e4-b4d4-556d7e7e9c35\"}]";
```

Comment:
There are some static methods on `PatchOperation` that will help you build a `JsonPatch` object explicitly rather than having to type out the JSON.